docs/architecture.md

```diff
@@ -29,73 +29,100 @@ have at least 3 nodes. In a basic setup with this amount of nodes,
-    instances before returning it completed back to the client. 
+    instances before returning it completed back to the client.
 
 Both replication types have their pros and cons.
 
 *Asynchronous replication* is faster. Also, you can distribute read and write
-requests of your application to different instances. From the other hand, it can
+requests of your application to different instances. On the other hand, it can
 be less reliable in terms of consistency: replicas may lag behind the primary
 instance, impacting any applications that depend on real-time data. Also, some
 transaction committed on the primary instance may not be available on replicas
 if the primary fails. Finally, asynchronous replication doesn't allow you
-scaling write requests horizontally, leaving  vertical scaling (primary
+scaling write requests horizontally, leaving vertical scaling (primary
 instance RAM and CPU increase) as the only available option.
 
-*Group replication* allows horizontal scaling of writes, but is not so fast,
-because writes require consensus (so-called "certification").
-Also, concurrent writes of the same row on different instances are
-[treated as data conflicts :octicons-link-external-16:](https://dev.mysql.com/doc/refman/8.0/en/group-replication-summary.html):
-the transaction that was ordered first commits on all servers, and the
-transaction ordered second aborts. Finally, group replication has some
+*Group replication* allows horizontal scaling of reads, but is slower. Group replication has some
 [specific limitations :octicons-link-external-16:](https://dev.mysql.com/doc/refman/8.0/en/group-replication-limitations.html).
 Particularly, the number of MySQL instances in a single replication group can't
-exceed 9. Also, extra large transaction can cause noticeable system slowdown,
+exceed 9. Also, extra large transactions can cause noticeable system slowdown,
 and in some cases even can cause the [replication member fault :octicons-link-external-16:](https://dev.mysql.com/doc/refman/8.0/en/group-replication-limitations.html#group-replication-limitations-transaction-size)
 when transaction message cannot be copied between group members over the
 network within a 5-second window.
 
+!!! note
+
+    MySQL documentation may also use the terms "source/replica" instead of
+    "primary/replica".
+
 Proxy solution used by the Operator for the database cluster depends on the
 replication type used in the cluster. [HAProxy](haproxy-conf.md) can be used
 with both replication types, while [MySQL Router](router-conf.md) can be used
 with Group Replication.
+
+### Replication and proxy comparison
+
+| Feature | Asynchronous replication + HAProxy | Group Replication + HAProxy/MySQL Router |
+| --- | --- | --- |
+| Writes | Single primary | Multi-primary possible (with consensus) |
+| Read scaling | Yes | Yes |
+| Write scaling | No | Limited (higher latency due to consensus) |
+| Consistency | Eventual on replicas | Stronger (group consensus) |
```

Comment:
Strong (depending the Transaction Consistency level see https://dev.mysql.com/doc/refman/8.4/en/group-replication-configuring-consistency-guarantees.html)